### PR TITLE
dynamic search placeholder, language and pin code support for the search

### PR DESCRIPTION
**Pull request by aakash-sonova**
[Link to Jira Ticket](https://sonovamarketing.atlassian.net/browse/ENGG-3112)

- Added props for the search input placeholder
- Updated search functionality to work for pin code
- Added functionality to search by contain (string `a` containing string `b`) to support search in other languages. The preference is to search by string matching with the starting letters, if not found we search with contain.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: build/js/intlTelInput-jquery.js [hunk 1]
```diff
@@ -935,7 +937,7 @@
                         _this9._clearInputField(e);
                     } else if (searchCountry === true && e.keyCode >= 96 && e.keyCode <= 105) {
                         _this9._searchCountry(e);
-                    } else if (/^[a-zA-ZÀ-ÿа-яА-Я ]$/.test(e.key)) {
+                    } else if (/^[a-zA-ZÀ-ÿа-яА-Я0-9 ]$/.test(e.key)) {
```
Comment (970dbarbato) [inline build/js/intlTelInput-jquery.js:940]: Why the numbers? Does it also search based on the country code number?

File: src/js/intlTelInput.js [hunk 2]
```diff
@@ -1054,19 +1057,35 @@ class Iti {
...
             listItem = this.countryList.querySelector("#iti-".concat(this.id, "__item-").concat(this.countries[i].iso2));
         } else if (this._startsWith(this.countries[i].dialCode, e.target.value)) {
             listItem = this.countryList.querySelector(`[data-dial-code='${this.countries[i].dialCode}'`);
         }
+
         // update highlighting and scroll
         if(listItem) {
-            this._highlightListItem(listItem, false);
-            this._scrollTo(listItem, true);
+            this._hilightAndScrollTo(listItem);
```
Comment (970gmena) [inline src/js/intlTelInput.js]: I'm not sure if this is a misspell of highlightAndScrollTo, if not ignore this comment

File: src/js/intlTelInput.js [hunk 3]
```diff
@@ -1054,19 +1057,35 @@ class Iti {
...
+    // If country not found in starts with and also in country code the checking for contains (For supporting other languages than english)
+    if (listItem === null) {
+      for (i = 0; i < this.countries.length; i++) {
+        if (this._contains(this.countries[i].name, e.target.value)) {
+            listItem = this.countryList.querySelector("#iti-".concat(this.id, "__item-").concat(this.countries[i].iso2));
+        }
+
+        // update highlighting and scroll
+        if(listItem) {
+          this._hilightAndScrollTo(listItem);
```
Comment (970gmena) [inline src/js/intlTelInput.js]: Same as last comment, if changing method name change in here as well

File: build/js/intlTelInput-jquery.js (hunk 1 above)
Comment (aakash-sonova) [reply build/js/intlTelInput-jquery.js:940]: Yes, Added this functionality too. :)
Or I would say the functionality was already there from pkg, but numbers were just not accepted in the input.

File: src/js/intlTelInput.js [hunk 4]
```diff
@@ -1079,7 +1098,18 @@ class Iti {
...
+    return a.substr(0, b.length).toLowerCase() === b.toLowerCase();
+  }
+
+  // check if string a contains string b
+  _contains(a, b) {
+    return a.toLowerCase().includes(b.toLowerCase());
+  }
+
+  // hilight and scroll to country
+  _hilightAndScrollTo(listItem) {
```
Comment (970gmena) [inline src/js/intlTelInput.js]: And also here and the line below

**Review by 970gmena — COMMENTED** [review]
I added some comments that don't interact with the functionality but maybe should be changed if it's a misspell error

**Review by 970dbarbato — APPROVED** [review]
LGTM, but left a comment

File: src/js/intlTelInput.js (hunk 2 above)
Comment (aakash-sonova) [reply src/js/intlTelInput.js]: Yes, it is, let me fix it. Thanks!

**Review by 970dbarbato — APPROVED** [review]
LGTM

**Pull request merged**